### PR TITLE
ffx feedstock

### PR DESCRIPTION
**Pull request by Ohjeah**

### CODE REVIEW

File: recipes/ecmwf-api-client/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name: ecmwf-api-client
+  version: "1.4"
+
+source:
+  fn: ecmwf-api-client-python.tgz
+  url: https://software.ecmwf.int/wiki/download/attachments/56664858/ecmwf-api-client-python.tgz
+  sha256: 423331f2ace3b247ff0d7db8c96952003136b5859ad0840a045c311ee857bcfb
+
+build:
+  number: 0
```
Comment (ocefpaf) [inline recipes/ecmwf-api-client/meta.yaml]: @Ohjeah you can add a line here with the install instructions.

```yaml
script: python setup.py install --single-version-externally-managed --record record.txt
```

Note that when the software uses setuptools we add the extra args `--single-version-externally-managed --record record.txt` to ensure the installed files play nice with external packaging tools.

File: recipes/ecmwf-api-client/build.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+$PYTHON setup.py install
+
+# Add more build steps here, if they are necessary.
+
+# See
+# http://docs.continuum.io/conda/build.html
+# for a list of environment variables that are set during the build process.
```
Comment (ocefpaf) [inline recipes/ecmwf-api-client/build.sh]: Hi @Ohjeah you don't need the `bld.bat` nor the `build.sh` here. (See https://github.com/conda-forge/staged-recipes/pull/685/files#r64913933)

File: recipes/ffx/build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+$PYTHON setup.py install
+
+# Add more build steps here, if they are necessary.
+
+# See
+# http://docs.continuum.io/conda/build.html
+# for a list of environment variables that are set during the build process.
```
Comment (ocefpaf) [inline recipes/ffx/build.sh]: Same thing as above. You can remove the `bld.bat` and `build.sh`, and then add the `script` key line in the `meta.yaml` file `build` sections.

File: recipes/ecmwf-api-client/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package:
+  name: ecmwf-api-client
+  version: "1.4"
+
+source:
+  fn: ecmwf-api-client-python.tgz
+  url: https://software.ecmwf.int/wiki/download/attachments/56664858/ecmwf-api-client-python.tgz
```
Comment (ocefpaf) [inline recipes/ecmwf-api-client/meta.yaml]: I don't understand the `ecmwf` software stack, so I am curious how does this package relates to: 

- https://github.com/conda-forge/ecmwf_grib-feedstock
- https://github.com/conda-forge/python-ecmwf_grib-feedstock
- https://github.com/conda-forge/eccodes-feedstock

Also, would you be interested in help maintaining any of the above?

**Comment by Ohjeah** [comment]
Hi @ocefpaf

thanks for the remarks. I will update the meta.yaml as you suggested. 

ecmwf-api-client lets you send requests to the ecmwf servers to download their data.
The shell tool ecmwf_grib is needed to read the binary format. python-ecmwf are the corresponding python bindings. I have not used eccodes myself, but it seems that it is another binary file parser which can handle more formats.

I am not a developer of those tools, just a user. Nevertheless I am willing to help maintaining the recipes.

**Comment by ocefpaf** [comment]
Awesome. And thanks for the explanation. You can add yourself as maintainer on the feedstocks that you wish by sending a PR adding your GitHub handle in the maintainer list. Once merge you'll have commit rights to the repo.

**Comment by ocefpaf** [comment]
I cannot merge a failing PR on the phone, but this is good to go as the failure is just a service time out. Can someone else either restart AppVeyor or merge this?

File: recipes/ffx/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+       - ffx
+
+about:
+  home: https://github.com/natekupp/ffx
+  license: See readme
+  summary: 'Fast Function Extraction: A fast, scalable, and deterministic symbolic regression tool.'
+
+extra:
+  recipe-maintainers:
+    Ohjeah
```
Comment (jakirkham) [inline recipes/ffx/meta.yaml]: `Ohjeah` -> `- Ohjeah`

**Comment by Ohjeah** [comment]
Hi, is this ready to be accepted or is there anything left for me to do?

File: recipes/ffx/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    - scipy
+    - pandas
+test:
+
+  imports:
+       - ffx
+
+about:
+  home: https://github.com/natekupp/ffx
+  license: See readme
```
Comment (ocefpaf) [inline recipes/ffx/meta.yaml]: You'll need to add more metadata, similar to [this](https://github.com/conda-forge/zlib-feedstock/blob/master/recipe/meta.yaml#L32-L35), here.

Example:

```yaml
    license: ffx
    license_family: Other
    license_file: LICENSE
```

You many need to copy the [license](https://github.com/natekupp/ffx/blob/master/LICENSE) file into the recipe directory if the tarball does not ship it.

**Comment by ocefpaf** [comment]
@Ohjeah there is a remaining issue with the license. Once that is fixed we can merge this ASAP.

**Comment by Ohjeah** [comment]
Thanks for the reminder @ocefpaf

File: recipes/ffx/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    git_tag: d1f027ed5f4a543c41af13f4b9a2be882ec4dd6a
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
```
Comment (ocefpaf) [inline recipes/ffx/meta.yaml]: The Linux docker image changed and some tools were removed. Can you add `git` to the build requirements? Or better yet. Change,

```yaml
source:
    git_url: https://github.com/Ohjeah/ffx
    git_tag: d1f027ed5f4a543c41af13f4b9a2be882ec4dd6a
```

to (the first line at the top right below the version)

```yaml
{% set commit = "d1f027ed5f4a543c41af13f4b9a2be882ec4dd6a" %}

source:
    fn: {{ commit }}.tar.gz
    url: https://github.com/Ohjeah/ffx/archive/{{ commit }}.tar.gz
    sha256: 44977347d9c9a60838283e3a684efe10a7772129c7d604fdc6695e85b9d7fc1f

```

**Comment by ocefpaf** [comment]
Thanks @Ohjeah!

**Pull request merged**